src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -302,6 +302,23 @@ ss::future<ss::stop_iteration> leader_balancer::balance() {
         }
     }
 
+    if (_in_flight_changes.size() >= transfer_limit_per_shard * cores.size()) {
+        vlog(
+          clusterlog.debug,
+          "Leadership balancer tick: number of in flight changes is at max "
+          "allowable"
+          ": {}. Retrying in {} seconds",
+          _in_flight_changes.size(),
+          std::chrono::duration_cast<std::chrono::seconds>(_mute_timeout())
+            .count());
+        // We wait for _mute_timeout here since it's the timeout for in flight
+        // requests.
+        if (!_timer.armed()) {
+            _timer.arm(_mute_timeout());
```

Comment:
Mute timeout is 5 minute by default -- we probably want a much shorter delay?

Reply:
Changed the timeout to be the shortest duration to when the next in flight change expires.

Reply:
Hmm, but that expiry time is initialized as now()+mute_timeout, so typically it's still going to be really long. The usual-case time when the entry from _in_flight_changes clears is much sooner (when we get a call to `on_leadership_change` for that ntp) rather than waiting for that expiry.

If we are on a system that's progressing well, then I expect we will quickly hit our limit, but then also a movement will finish a few second later.  The sleep in this case should reflect our expectation of how long we think a leader transfer takes (i.e. how long to wait before a slot will be free).

Maybe we can make this a bit smarter by sleeping on the idle timeout here, but also setting a "_throttled=true" flag, then having on_leadership_change  do a `_timer.rearm` with a much shorter time when it erases something from _in_flight_changes and sees _throttled==true.